packages/webpack/src/types.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @typedef {Object} Policy
+ * @property {Record<string,object>} resources
+ */
+
+/**
+ * @typedef {Object} LavaMoatPluginOptions
+ * @property {boolean} [runChecks] - check resulting code with wrapping for correctness
+ * @property {boolean} [readableResourceIds] - should resourceIds be readable or turned into numbers - defaults to (mode==='development')
+ * @property {number} [diagnosticsVerbosity] - a number representing diagnostics output verbosity, the larger the more overwhelming
+ * @property {Object} policy - LavaMoat policy
+ * @property {Object} [lockdown] - options to pass to lockdown
+ */
+
+/**
+ * @typedef {function} GetEndowmentsForConfig Creates an object populated with only the deep properties specified in the packagePolicy
+ * @param {object} sourceRef - Object from which to copy properties
+ * @param {object} packagePolicy - LavaMoat policy item representing a package
+ * @param {object} unwrapTo - For getters and setters, when the this-value is unwrapFrom, is replaced as unwrapTo
+ * @param {object} unwrapFrom - For getters and setters, the this-value to replace (default: targetRef)
+ * @return {object} - The targetRef
+ *
+ */
+
+/**
+ * @typedef {Object} EndowmentsToolkit
+ * @property {GetEndowmentsForConfig} getEndowmentsForConfig
+ * @property {Function} makeMinimalViewOfRef - Creates a minimal view of a reference (e.g. a global) that only exposes the properties specified in the packagePolicy
+ * @property {Function} copyValueAtPath - Copies a value at a specific path from the source reference to the target reference
+ * @property {Function} applyGetSetPropDescTransforms 
+ * @property {Function} applyEndowmentPropDescTransforms 
+ * @property {Function} copyWrappedGlobals - Copies wrapped globals from a global reference to a target object with wrapping
+ * @property {Function} createFunctionWrapper - conditionally binds a function - used to ensure functions (like fetch) get called on the this they expect
+ */
+
+/**
+ * @typedef {function} EndowmentsToolkitFactory
```

Comment:
```suggestion
 * @callback EndowmentsToolkitFactory
```

I actually have types for this stuff in `endo-refactor` branch.  I should probably extract them and send a PR